lib/streamlit/state/widgets.py

```diff
@@ -268,16 +271,14 @@ def _get_widget_id(
 ) -> str:
     """Generate a widget id for the given widget.
 
-    If user_key is defined, the widget_id returned is simply user_key.
-    Otherwise, we return a hash of the widget element type and the
-    string-serialized widget proto.
+    The widget id includes the user_key, which may be `None`, so widgets with
+    identical arguments can use it to be distinct.
+    The widget id includes an easily identified prefix, and the user_key as a
+    suffix, to make it easy to identify it and know if a key maps to it.
```

Comment:
```suggestion
    The user_key is included in the widget id allowing widgets with identical arguments to have distinct ids. Note: user_key may be `None`. 
    The widget id includes an easily identified prefix, and the user_key as a
    suffix, to make it easy to identify it and know if a key maps to it.
```

Having the info about user_key possibly being `None` as a clause in the sentence made it less readable for me. Something like this might be a bit clearer.